sdks/python/apache_beam/io/hadoopfilesystem.py

```diff
@@ -188,13 +194,16 @@ def _match(path_pattern, limit):
       """Find all matching paths to the pattern provided."""
       fs = self._hdfs_client.status(path_pattern, strict=False)
       if fs and fs[_FILE_STATUS_TYPE] == _FILE_STATUS_TYPE_FILE:
-        file_statuses = [(fs[_FILE_STATUS_PATH_SUFFIX], fs)][:limit]
+        file_statuses = [(path_pattern, fs)][:limit]
       else:
-        file_statuses = self._hdfs_client.list(path_pattern,
-                                               status=True)[:limit]
-      metadata_list = [FileMetadata(file_status[1][_FILE_STATUS_NAME],
-                                    file_status[1][_FILE_STATUS_SIZE])
-                       for file_status in file_statuses]
+        file_statuses = [(self._join(path_pattern, fs[0]), fs[1])
+                         for fs in self._hdfs_client.list(path_pattern,
```

Comment:
Does list operation here handle common glob patterns or is it just '*' ? (we should document if latter).

Reply:
No glob patterns are supported at all. If a directory is passed it'll list its contents.

Reply:
I did start adding glob support in Feb, but somehow forgot about it.
https://issues.apache.org/jira/browse/BEAM-4011